File: sol/solution/0201-0300/0203/Solution.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */
function removeElements(head: ListNode | null, val: number): ListNode | null {
    const dummy = new ListNode(0, head);
    let curr: ListNode | null = dummy;
    while (curr && curr.next) {
        if (curr.next.val === val)
            curr.next = curr.next.next;
        else
            curr = curr.next;
    }
    return dummy.next;
}

File: sol/solution/0201-0300/0203/Solution.rb
# Definition for singly-linked list.
# class ListNode
#     attr_accessor :val, :next
#     def initialize(val = 0, _next = nil)
#         @val = val
#         @next = _next
#     end
# end
# @param {ListNode} head
# @param {Integer} val
# @return {ListNode}
def remove_elements(head, val)
    dummy = ListNode.new(0)
    dummy.next = head
    curr = dummy
    while curr.next
        if curr.next.val == val
            curr.next = curr.next.next
        else
            curr = curr.next
        end
    end
    dummy.next
end